cli/context/store/io_utils.go

```diff
@@ -0,0 +1,58 @@
+package store
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// limitedReader a wrapper on io.Reader for error handling of ReadAll
+type limitedReader struct {
```

Comment:
@goksu I don't understand why you're implementing your own limitReader. Isn't passing a reader to https://godoc.org/io#LimitReader sufficient?

Reply:
Hey Tibor, passing the reader through LimitReader is in fact sufficient. But, there was an earlier comment on the PR from @thaJeztah indicating the fact that the error handling on the reader coming back from `LimitReader` falling short on signifying to the user that the failure is in fact from file size being more than allowed amount.

So instead of failing silently with an `EOF`, I've went with this route per recommendation. If we think that is an overkill, then I can happily revert that section and just wrap the reader with a `LimitReader`.

Reply:
The custom version was to allow producing a more useful error message https://github.com/docker/cli/pull/1895#discussion_r286110269

Reply:
Ah, sorry; our comments crossed

Reply:
@tiborvass — With the above comments giving you more context, are you okay with keeping the implementation as is, or would you still recommend the simpler route of just using the `LimitReader`.